### PR TITLE
Update default team behaviour

### PR DESCRIPTION
**Pull request by quentingrchr**
## We switch to a cookie based default team behaviour
Navigation to `/teams/...` will set the default team in the cookie. Then navigating to `/teams` will redirect to the default team based on that cookie, if no cookie is set it will redirect to the first team found.

### CODE REVIEW

**Comment by quentingrchr** [comment]
On attend un fix potentiel sur une issue Next pour utiliser middleware + cookie
Liée aussi à cette [issue](https://github.com/premieroctet/reactivated-app-v2/issues/236)

**Comment by quentingrchr** [comment]
TLDR:
- Setting the cookie within a client component
- Redirecting based on the cookie within a server component
- Opting out of caching on /teams/:slug to keep the cookie up to date


## Update : 
We've switched to client-side cookie setting because cookies can only be set in route handlers or server actions (and we encountered issues with server actions blocking navigation).

Additionally, we had to do without using the `<NextLink/>` component for navigation between teams because Next.js, by default, has an overly aggressive caching policy, and the page code was never re-executed after the initial caching (despite various attempts like:

```typescript
export const fetchCache = 'force-no-store'
export const revalidate = 0 // seconds
export const dynamic = 'force-dynamic'
```
